include/odp/api/spec/proto_stats.h

```diff
@@ -0,0 +1,205 @@
+ * By default all the statistics are disabled.
+ *
+ * @param param   Proto stats parameter pointer.
+ */
+void odp_proto_stats_param_init(odp_proto_stats_param_t *param);
+
+/**
+ * Get proto stats capability
+ *
+ * Get supported protocol statistics and metadata.
+ *
+ * @param capa Pointer where capabilities are updated.
+ */
+void odp_proto_stats_capability(odp_proto_stats_capability_t *capa);
+
+/**
+ * Create a proto stats object
+ *
+ * Create a proto stats object with given name and parameters.
+ *
+ * @param name  Object name
+ * @param param Proto stats parameters
+ *
+ * @return Proto stats object handle
+ * @retval ODP_PROTO_STATS_INVALID on failure
+ */
+odp_proto_stats_t odp_proto_stats_create(const char *name, const odp_proto_stats_param_t *param);
+
+/**
+ * Lookup a proto stats object by name
+ *
+ * Lookup an already created proto stats object by name.
+ *
+ * @param name Proto stats object name
+ *
+ * @return Proto stats object handle
+ * @retval ODP_PROTO_STATS_INVALID on failure
+ */
+odp_proto_stats_t odp_proto_stats_lookup(const char *name);
+
+/**
+ * Destroy a proto stats object
+ *
+ * Destroy a proto stats object already created.
+ *
+ * - Before destroying proto stats object having ODP_PROTO_STATS_ID_TX_*
+ *   enabled, for all PKTIO devices to which packets were Tx'ed earlier with
+ *   this proto stats object, odp_pktio_stop() must be called. Additionally,
+ *   existing packets that refer to the proto stats object being destroyed
+ *   must not be sent at the same time as or after the proto stats object
+ *   destruction.
+ *
+ * @param stat Proto stats handle
+ *
+ * @retval 0 on success
+ * @retval <0 on failure
+ */
+int odp_proto_stats_destroy(odp_proto_stats_t stat);
+
+/**
+ * Get proto stats counter by id
+ *
+ * Get current value of statistics counter by stat id.
+ *
+ * @param stat Proto stat object handle
+ * @param id   Stats counter identifier to return the value.
+ *
+ * @return Stats counter value
+ * @retval Unknown value if stats id is not valid for the object.
+ */
+uint64_t odp_proto_stats(odp_proto_stats_t stat, odp_proto_stats_id_t id);
+
+/** ODP proto stats counters */
+typedef struct odp_proto_stats_data_t {
+	/** Array of counters */
+	uint64_t counter[ODP_PROTO_STATS_ID_MAX];
+} odp_proto_stats_data_t;
+
+/**
+ * Get all proto stats counters
+ *
+ * Get current values of all counters of the proto stats object.
+ * The values of counters that are not enabled in the proto stats object are undefined.
+ *
+ * @param      stat   Proto stats object handle
+ * @param[out] data   Pointer to a caller allocated structure where the statistics will
+ *                    be written to.
+ *
+ * @retval =0 on success
+ * @retval <0 on failure
+ */
+int odp_proto_stats_all(odp_proto_stats_t stat, odp_proto_stats_data_t *data);
+
+/**
+ * Print proto stats object info on console.
+ *
+ * Print implementation-defined proto stats debug information to the console.
```

Comment:
These debug print functions print "to the ODP log", not to the console. Application may override the log function, and that function may not output to the console, but into file, etc.